apps/aecore/test/aecore_persistence_test.exs

```diff
@@ -3,30 +3,63 @@ defmodule PersistenceTest do
   doctest Aecore.Persistence.Worker
 
   alias Aecore.Persistence.Worker, as: Persistence
+  alias Aecore.Keys.Worker, as: Keys
   alias Aecore.Miner.Worker, as: Miner
   alias Aecore.Chain.Worker, as: Chain
   alias Aecore.Chain.BlockValidation
 
-  setup do
+  setup persistance_state do
     Persistence.start_link([])
     Miner.start_link([])
-    []
+    Miner.mine_sync_block_to_chain
+    path = Application.get_env(:aecore, :persistence)[:path]
+    on_exit fn ->
+      File.rm_rf(path)
```

Comment:
Hmm, that seems dangerous. I'm not sure what we can do, but some error in `:path` env variable can be catastrophic. Maybe we could check on previous line if the folder is empty and fail the test if it's not?

Reply:
I would do: `assert false == File.exists?(path)`

Reply:
I'd check if the path is kind of what we expect

Reply:
My suggestion doesn't work. The path already exists when this test is invoked. If you won't have any idea how to fix this we can merge as is.

Reply:
@meivantodorov any ideas?

Reply:
I will simply add `if do` and we can finish this